plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java

```diff
@@ -141,8 +140,11 @@ private void handleSelectedAppSource(PropertyChangeEvent evt) {
 		if(evt.getNewValue() instanceof IApplicationSource
 				&& ResourceKind.TEMPLATE.equals(((IApplicationSource) evt.getNewValue()).getKind())){
 			IApplicationSource source = (IApplicationSource) evt.getNewValue();
-			this.template = (ITemplate) source.getSource();
-			updateTemplateParameters(template);
+			ITemplate newTemplate = (ITemplate) source.getSource();
+			if (!Objects.equals(newTemplate, this.template)) {
```

Comment:
dont update template parameters if we already have the very same template already set